### PR TITLE
5912 multiple credentials

### PR DESCRIPTION
**Pull request by mscottTU**
Code remains unchanged for users who wish to continue using just one set of credentials.

Users who wish to use more than 1 set of credentials just need to call ApiClient.SetUp(userName, password, secret, baseUrl) for each MerchantSite, then add the username for the MerchantSite they wish to make Api calls on as a parameter to each of their method calls. For example:

**Before**:
```
ApiClient.SetUp(userName1, password1, secret1, baseUrl1);

ApiClient.postCase(kase1);
```

**After**:
```
ApiClient.SetUp(userName1, password1, secret1, baseUrl1);
ApiClient.SetUp(userName2, password2, secret2, baseUrl2);

ApiClient.postCase(kase1, userName1);
ApiClient.postCase(kase2, userName2);
```

### CODE REVIEW

File: src/main/java/com/trustev/web/ApiClient.java [hunk 1]
```diff
@@ -22,727 +21,1387 @@
...
+     */
+    public static Email getEmail(String caseId, String emailId, String username) throws TrustevApiException {
+        String url = "/Case/{id}/Customer/Email/{id2}".replace("{id}", caseId).replace("{id2}", emailId);
+        ;
+        Email response = (Email) PerformHttpCall(url, HttpMethod.GET, Email.class, null, true, username);
+        return response;
+    }
+
+    /**
+     * Get all the Emails from
```
Comment (ufranca1984) [inline src/main/java/com/trustev/web/ApiClient.java]: we can get the username and call PerformHttpCall(uriPath, httpMethod, responseType, entity, isAuthenticationNeeded, userName)

File: src/main/java/com/trustev/domain/entities/MerchantSite.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    @JsonProperty("Secret")
+    public void setSecret(String secret) {
+        this.secret = secret;
+    }
+
+    public String getBaseUrlString() {
+        return baseUrlString;
+    }
+
+    @JsonProperty("baseUrlString")
```
Comment (ufranca1984) [inline src/main/java/com/trustev/domain/entities/MerchantSite.java]: Upper case BaseUrlString on the property

File: src/main/java/com/trustev/domain/entities/MerchantSite.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    public void setBaseUrlString(String baseUrlString) {
+        this.baseUrlString = baseUrlString;
+    }
+
+    public BaseUrl getBaseUrl() {
+        return baseUrl;
+    }
+
+    @JsonProperty("baseUrl")
+    public void setBaseUrl(BaseUrl baseUrl) {
```
Comment (ufranca1984) [inline src/main/java/com/trustev/domain/entities/MerchantSite.java]: Upper case BaseUrl ...

**Review by ufranca1984 — COMMENTED** [review]
Just 3 points.

File: src/main/java/com/trustev/web/ApiClient.java [hunk 4]
```diff
@@ -22,727 +21,1339 @@
...
+     */
+    public static boolean hasMultipleMerchantSites() {
+        if (merchantSites != null && merchantSites.size() > 1) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public static void removeAllMerchantSites() {
```
Comment (csweetn) [inline src/main/java/com/trustev/web/ApiClient.java]: missing description

File: src/main/java/com/trustev/web/ApiClient.java [hunk 5]
```diff
@@ -22,727 +21,1339 @@
...
+        merchantSites.put(userName, new MerchantSite(userName, password, secret, baseUrl));
+    }
+
+    /**
+     * Check if the Merchant Site associated with the username is already in the Map of merchantSites
+     *
+     * @param userName the username for the given merchant site
+     * @return true if the merchant site is already registered, false if the merchant site is not registered (or if the map of merchantSites is empty)
+     */
+    public static boolean hasMerchantSite(String userName) {
```
Comment (csweetn) [inline src/main/java/com/trustev/web/ApiClient.java]: Should some of these merchantSite methods be made private?? Do we want to expose all this functionality??

**Review by csweetn — COMMENTED** [review]
Just two small things to consider from me!! We also need to update the README to show this new functionality. Not sure if this is integrations job though...

**Review by JFreitas85 — APPROVED** [review]
Looks good!

**Review by jfreitasTrustev — APPROVED** [review]

**Pull request merged**